pkg/operator/ceph/object/rgw.go

```diff
@@ -350,8 +350,16 @@ func getDomainName(s *cephv1.CephObjectStore, returnRandomDomainIfMultiple bool)
 			idx = rand.Intn(len(endpoints)) //nolint:gosec // G404: cryptographically weak RNG is fine here
 		}
 		return endpoints[idx].String()
-	}
+	} else if len(s.Spec.Hosting.DNSNames) > 0 {
+		// if the store is internal and has DNS names, pick a random DNS name to use. if the DNS name
+		endpoints := s.Spec.Hosting.DNSNames
+		idx := 0
 
+		if returnRandomDomainIfMultiple {
+			idx = rand.Intn(len(endpoints)) //nolint:gosec // G404: cryptographically weak RNG is fine here
+		}
+		return endpoints[idx]
+	}
```

Comment:
There's quite a bit of code that is just duplicated between these branches. Let's combine logic so future edits are less likely to skew between configs.

```golang
endpoints := []string{}
if s.Spec.IsExternal() {
    endpoints = s.Spec.Gateway.ExternalRgwEndpoints
} else if len(s.Spec.Hosting.DNSNames) > 0 {
    endpoints = s.Spec.Hosting.DNSNames
} else {
   return domainNameOfService(s)
}
// TODO: rand logic
```